src/platform/android/DeviceInstanceInfoProviderImpl.cpp

```diff
@@ -73,6 +73,32 @@ CHIP_ERROR DeviceInstanceInfoProviderImpl::GetProductLabel(char * buf, size_t bu
     return Internal::AndroidConfig::ReadConfigValueStr(Internal::AndroidConfig::kConfigKey_ProductLabel, buf, bufSize, dateLen);
 }
 
+CHIP_ERROR DeviceInstanceInfoProviderImpl::GetDeviceName(MutableCharSpan & deviceNameSpan)
+{
+    CHIP_ERROR err;
+    size_t deviceNameSize = 0;
+    char androidDeviceName[ConfigurationManager::kMaxDeviceNameLen];
+    MutableCharSpan androidDeviceNameSpan(androidDeviceName);
+
+    err = Internal::AndroidConfig::ReadConfigValueStr(Internal::AndroidConfig::kConfigKey_MfrDeviceName,
+                                                      androidDeviceNameSpan.data(), androidDeviceNameSpan.size(), deviceNameSize);
+
+    if (err == CHIP_NO_ERROR)
+    {
+        ReturnErrorCodeIf(deviceNameSpan.size() < deviceNameSize, CHIP_ERROR_BUFFER_TOO_SMALL);
+        memcpy(deviceNameSpan.data(), androidDeviceNameSpan.data(), deviceNameSize);
+        deviceNameSpan.reduce_size(deviceNameSize);
+    }
+    else if (err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
+    {
+        ReturnErrorCodeIf(deviceNameSpan.size() < strlen(CHIP_DEVICE_CONFIG_DEVICE_NAME), CHIP_ERROR_BUFFER_TOO_SMALL);
```

Comment:
This, again, seems like 
```
return CopyCharSpanToMutableCharSpan(CharSpan::fromCharString(CHIP_DEVICE_CONFIG_DEVICE_NAME), deviceNameSpan);
```

would do the right thing.

Reply:
replaced!